fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

muwoo opened the pull request:
fix(windows): When I minimize a window by clicking on the taskbar icon on Windows, calling the `requestUserAttention` function does not trigger the taskbar icon to flash. #942

muwoo committed:
fix(windows): #942 after minimizing the window by clicking the taskbar icon, the `requestUserAttention` function does not take effect.

amrbashir commented on src/platform_impl/windows/window.rs:
```diff
@@ -874,8 +874,12 @@ impl Window {
   #[inline]
   pub fn request_user_attention(&self, request_type: Option<UserAttentionType>) {
     let window = self.window.clone();
+    let window_flags = self.window_state.lock().window_flags();
+    let is_minimized = window_flags.contains(WindowFlags::MINIMIZED);
```
let's move this lock and its check inside the if statement, so locking the mutex can be avoided if the active window is not this window, something like this.

```rs
if window.0 == active_window_handle {
    let window_flags = self.window_state.lock().window_flags();
    let is_minimized = window_flags.contains(WindowFlags::MINIMIZED);
    if !is_minimized {
      return
    }
}
```

muwoo replied:
Thanks for your suggestion, it has been modified

amrbashir commented on .changes/fix-user-attention-minimized.md:
```diff
@@ -0,0 +1,5 @@
+"tao": patch
+---
+
+Fix: after minimizing the window by clicking the taskbar icon, the `requestUserAttention` function does not take effect.
```
```suggestion
Fix `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

muwoo replied:
Thanks for your suggestion, it has been modified

muwoo committed:
fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

muwoo committed:
fix(windows): #942 after minimizing the window by clicking the taskbar icon, the `requestUserAttention` function does not take effect.

fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

Update src/platform_impl/windows/window.rs

amrbashir commented on src/platform_impl/windows/window.rs:
```diff
@@ -875,8 +875,14 @@ impl Window {
+      let window_flags = self.window_state.lock().window_flags();
+      let is_minimized = window_flags.contains(WindowFlags::MINIMIZED);
+      if !is_minimized {
+        return
```
```suggestion
        return;
```

amrbashir committed:
Update src/platform_impl/windows/window.rs

amrbashir commented:
Thank you

muwoo committed:
fix(windows): `Window::request_user_attention` not taking effect after minimizing the window by clicking the taskbar icon

muwoo committed:
Merge remote-tracking branch 'origin/dev' into dev

amrbashir commented:
Oh looks like you didn't sign your commits and I can't merge the PR unless they are signed. You need to setup [commit signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits), then you can sign past commit like [this](https://superuser.com/questions/397149/can-you-gpg-sign-old-commits/1123928#1123928) for example.

muwoo committed:
Merge branch 'dev' of github.com:muwoo/tao into dev

muwoo commented:
@amrbashir Okay, I'll try to submit a PR again
